### DISCUSSION TITLE
How do you validate a checkbox boolean in a valibot form schema?

### DISCUSSION THREAD

**Discussion by rikhall1515** [open-circle/valibot#247] [Q&A] [answered]
Hi I'm really loving valibot so far, but I have run into a roadblock.

The below code is from a Qwik project with modular forms installed. I've been following their documentation but they didn't include any examples for how to validate checkboxes (with the help of valibot), which left me in the dark.

Here below is the code:

```jsx
const EmailSchema = object({
  ...
  terms: boolean("You must agree to the terms of service before submitting an email.", [
       //I want to check if the boolean value of the terms of service checkbox is equal to true.
       //In other words, I want the checkbox to be required while still being of the
  ]),
});

type EmailForm = Input<typeof EmailSchema>;

export const useFormLoader = routeLoader$<InitialValues<EmailForm>>(() => {
  return {
    ...
    terms: false,
  };
});
```
Based on having perused the documentation and looking at the code, I initially found ``equal`` which is deprecated, then ``value`` but that returns a ``PipeResult<string | number | bigint>`` which is not what I want.

I think fundamentally I have a misunderstanding here and I'm curious what the solution is to the goal I want to achieve😅.

Thanks in advance!

**Comment by fabian-hiller** [comment, marked as answer]
Thanks for the info. We should probably update `value` to support boolean as well. For now, `literal` is the way to go. It keeps the bundle size to a minimum and does a straight check for `true`. 

```ts
const Schema = literal(true, 'Please accept our terms.');
```

**Reply by rikhall1515** [reply]
Thank you so much for the speedy response. I think I might have totally missed that function.

I happened to find a way to make a custom function but what you showed here already does the same thing.

**Comment by adrianrudnik** [comment]
Just had the same issue. As a note: While the `literal()` approach is working, it's type does not seem to be compatible (not that it leads to actual errors, just typescript warnings), coming from a [Nuxt UI example](https://ui3.nuxt.dev/components/form#schema-validation).

```js
const formsSchema = v.object({
  accept_terms: v.literal(true),
})

type Schema = v.InferOutput<typeof formsSchema>
```

seems to lead to a `Schema`  where `accept_terms` infers type `true`, while the form value itself can be `boolean` with initial state of `false`. Not sure why or how, hoped for a `.true()` validation, that allows a boolean, but expects a truthy value as a valid value.

**Reply by fabian-hiller** [reply]
Does `v.pipe(v.boolean(), v.value(true))` works for you?

**Reply by adrianrudnik** [reply]
Seems to be working fine, thank you!
